### PR TITLE
Add Loopring 3.6

### PR DESCRIPTION
**Pull request by Brechtpd**
I've checked that:

* [x] the query produces the intended results
* [x] the folder name matches the schema name
* [ ] the schema name exists in Dune
* [x] views are prefixed with `view_`, functions with `fn_`.
* [x] the filename matches the defined view, table or function and ends with .sql
* [x] each file has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

### CODE REVIEW

File: schema/loopring/fn_decode_float_24.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+DROP FUNCTION loopring.fn_decode_float_24;
+
+CREATE OR REPLACE FUNCTION loopring.fn_decode_float_24(data bytea) RETURNS double precision AS $$
```
Comment (mewwts) [inline schema/loopring/fn_decode_float_24.sql:3]: I think this exists as `fn_decode_float`, already? Can we delete that and use this one the old block_processing?

File: schema/loopring/fn_process_block.sql [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    withdraw loopring.withdraw_struct,
+    transfer loopring.transfer_struct,
+    spot_trade loopring.spot_trade_struct,
+    account_update loopring.account_update_struct,
+    amm_update loopring.amm_update_struct,
+    signature_verification loopring.signature_verification_struct
+);
+
+
+CREATE OR REPLACE FUNCTION loopring.fn_process_block(blockSize integer, _data bytea, block_timestamp timestamptz, blockIdx integer)
```
Comment (mewwts) [inline schema/loopring/fn_process_block.sql:84]: Should we rename the "old" process_block function `loopring.fn_process_trade_block` to something that indicates a version?

File: schema/loopring/fn_process_block.sql [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+RETURNS loopring.transaction_struct[]
+AS $$
+DECLARE
+    data bytea;
+    i integer;
+    transaction loopring.transaction_struct;
+    transactions loopring.transaction_struct[];
+BEGIN
+
+    FOR i IN 0 .. blockSize-1
```
Comment (mewwts) [inline schema/loopring/fn_process_block.sql:94]: This for loop can be more efficient if replaces by generate_series like we did https://github.com/duneanalytics/abstractions/blob/master/schema/loopring/fn_process_trade_block.sql#L42-L43. 

I can do that when I merge :+1:

**Review by mewwts — COMMENTED** [review]
Looks good. Can you take a look at my comments?

File: schema/loopring/fn_process_block.sql (hunk 3 above)
Comment (Brechtpd) [reply schema/loopring/fn_process_block.sql]: Tried to do that previously, but failed. Will try again because I can't remember exactly why I had problems.

Comment (Brechtpd) [reply schema/loopring/fn_process_block.sql]:
```

CREATE OR REPLACE FUNCTION dune_user_generated.fn_process_transfer_block(blockSize integer, _data bytea, block_timestamp timestamptz, blockIdx integer) 
    RETURNS SETOF dune_user_generated.transaction_struct
    AS $$
DECLARE
    data bytea;
BEGIN
    -- Needs to depend on i
    data = ((substr(_data, 99/*+i*29*/, 29) || substr(_data, 99+blockSize*29/*+i*39*/, 39)));
    RETURN query
    SELECT
        block_timestamp,
        blockIdx,
        i,
        get_byte(substr(data, 1, 1), 0) as txType,
        (
            substr(data, 2, 20),
            dune_user_generated.fn_to_uint32(substr(data, 2 + 20, 4)),
            dune_user_generated.fn_to_uint16(substr(data, 2 + 20 + 4, 2)),
            dune_user_generated.fn_to_uint96(substr(data, 2 + 20 + 4 + 2, 12))
        ) as deposit,
        (
            substr(data, 2 + 1, 20),
            dune_user_generated.fn_to_uint32(substr(data, 2 + 1 + 20, 4)),
            dune_user_generated.fn_to_uint16(substr(data, 2 + 1 + 20 + 4, 2)),
            dune_user_generated.fn_to_uint96(substr(data, 2 + 1 + 20 + 4 + 2, 12)),
            dune_user_generated.fn_to_uint16(substr(data, 2 + 1 + 20 + 4 + 2 + 12, 2)),
            dune_user_generated.fn_decode_float_16(substr(data, 2 + 1 + 20 + 4 + 2 + 12 + 2, 2))
        ) as withdraw,
        (
            dune_user_generated.fn_to_uint16(substr(data, 2 + 1 + 4 + 4, 2)),
            dune_user_generated.fn_decode_float_24(substr(data, 2 + 1 + 4 + 4 + 2, 3)),
            dune_user_generated.fn_to_uint16(substr(data, 2 + 1 + 4 + 4 + 2 + 3, 2)),
            dune_user_generated.fn_decode_float_16(substr(data, 2 + 1 + 4 + 4 + 2 + 3 + 2, 2)),
            dune_user_generated.fn_to_uint32(substr(data, 2 + 1, 4)),
            dune_user_generated.fn_to_uint32(substr(data, 2 + 1 + 4, 4)),
            substr(data, 2 + 1 + 4 + 4 + 2 + 3 + 2 + 2 + 4, 20),
            substr(data, 2 + 1 + 4 + 4 + 2 + 3 + 2 + 2 + 4 + 20, 20)
        ) as transfer,
        (
            dune_user_generated.fn_to_uint32(substr(data, 2 + 4 + 4, 4)),
            dune_user_generated.fn_to_uint32(substr(data, 2 + 4 + 4 + 4, 4)),
            dune_user_generated.fn_to_uint16(substr(data, 2 + 4 + 4 + 4 + 4, 2)),
            dune_user_generated.fn_to_uint16(substr(data, 2 + 4 + 4 + 4 + 4 + 2, 2)),
            dune_user_generated.fn_decode_float_24(substr(data, 2 + 4 + 4 + 4 + 4 + 2 + 2, 3)),
            dune_user_generated.fn_decode_float_24(substr(data, 2 + 4 + 4 + 4 + 4 + 2 + 2 + 3, 3))
        ) as spot_trade,
        (
            substr(data, 2 + 1, 20),
            dune_user_generated.fn_to_uint32(substr(data, 2 + 1 + 20, 4)),
            dune_user_generated.fn_to_uint16(substr(data, 2 + 1 + 20 + 4, 2)),
            dune_user_generated.fn_decode_float_16(substr(data, 2 + 1 + 20 + 4 + 2, 2)),
            substr(data, 2 + 1 + 20 + 4 + 2 + 2, 32)
        ) as account_update,
        (
            substr(data, 2, 20),
            dune_user_generated.fn_to_uint32(substr(data, 2 + 20, 4)),
            dune_user_generated.fn_to_uint16(substr(data, 2 + 20 + 4, 2)),
            get_byte(substr(data, 2 + 20 + 4 + 2, 1), 0),
            dune_user_generated.fn_to_uint96(substr(data, 2 + 20 + 4 + 2 + 1, 12))
        ) as amm_update,
        (
            substr(data, 2, 20),
            dune_user_generated.fn_to_uint32(substr(data, 2 + 20, 4))
        ) as signature_verification
    FROM
        generate_series(0, blockSize - 1) i;
END;
$$
LANGUAGE PLPGSQL;
```

2 problems I don't know how to fix:
- `data` depends on i, but I don't know how to do this without simply copying the whole `((substr(_data, 99/*+i*29*/, 29) || substr(_data, 99+blockSize*29/*+i*39*/, 39)))` everywhere `data` is used in `SELECT`
- `Error running query: structure of query does not match function result type DETAIL: Returned type record does not match expected type dune_user_generated.deposit_struct in column 5. CONTEXT: PL/pgSQL function dune_user_generated.fn_process_transfer_block(integer,bytea,timestamp with time zone,integer) line 6 at RETURN QUERY`. I don't know what the problem is here. I tried to add `AS` for the deposit members, but that isn't allowed.

Comment (mewwts) [reply schema/loopring/fn_process_block.sql]: Thanks @Brechtpd I'll have a look!

**Comment by mewwts** [comment]
Closing in favor of https://github.com/duneanalytics/abstractions/pull/205

**Pull request closed without merging**